src/components/HomePage/ReadingStreak/index.tsx

```diff
@@ -0,0 +1,242 @@
+/* eslint-disable max-lines */
```

Comment:
I feel this file needs to be broken down into multiple components as there is a lot of conditional rendering